pipeline/custom.groovy

```diff
@@ -0,0 +1,164 @@
+		'suite_name' : "suites/${params.Ceph_Version}/${params.Group}/${params.Suite}.yaml",
+		'global_conf' : "conf/${params.Ceph_Version}/${params.Group}/${params.Conf}.yaml",
+		'instances_name' : instances_name,
+		'addn_args': "${params.Additional_Args}"
+	]
+    def git_branch = "${params.Branch}"
+    def repo_link = "${params.Git_Repo}"
+    def destroy_cluster = "${params.Destroy_Cluster}"
+    def cli = GetCLI(args)
+    String stageName = "Stage ${params.Suite}"
+    stage(stageName){
+        def dirName = UUID.randomUUID().toString()
+        def rc = -1
+        def ret = -1
+        dir(dirName){
+            echo 'Executing CLI ' + cli
+            checkout([$class: 'GitSCM',
+                branches: [[name: git_branch]],
+                doGenerateSubmoduleConfigurations: false,
+                extensions: [],
+                submoduleCfg: [],
+                userRemoteConfigs: [[url: repo_link]]
+            ])
+            echo "Check out complete"
+            try {
+                rc = sh(script: cli, returnStatus: true)
+            }
+            catch(Exception ex){
+                echo 'Exception occured while executing the suite'
+                echo ex.getMessage()
+            }
+            finally{
+                echo "Execution completed"
+                echo rc.toString()
+                try {
+                    if((destroy_cluster == 'Destroy when suite/s pass' && rc == 0) || (destroy_cluster == 'Destroy always')){
+                        sh 'python3 -m venv venv;source venv/bin/activate; python run.py --cleanup=' + instances_name + ' --osp-cred ~/osp-cred-ci-2.yaml --log-level info'
+                    }
+                    ret = 0
+                }
+                catch(Exception ex){
+                    echo 'Exception occured while cleaning up the nodes'
+                    echo ex.getMessage()
+                    ret = -1
+                }
+                echo "Cleanup completed"
+            }
+        }
+        def wsd = env.WORKSPACE
+        echo wsd
+        sh 'rm -r ' + wsd + '/' + dirName
+        catchError(buildResult: 'SUCCESS', stageResult: 'FAILURE') {
+            if(rc != 0 || ret != 0){
+                sh "exit 1"
+            }
+        }
+    }
+}
+
+node(nodeName){
+    stage('Install pre req') {
+        if (env.WORKSPACE) {
+            sh script: "sudo rm -rf *"
+        }
+        checkout([
+            $class: 'GitSCM',
+            branches: [[name: '*/custom_jenkins_job']],
+            doGenerateSubmoduleConfigurations: false,
+            extensions: [[
+                $class: 'SubmoduleOption',
+                disableSubmodules: false,
+                parentCredentials: false,
+                recursiveSubmodules: true,
+                reference: '',
+                trackingSubmodules: false
+            ]],
+            submoduleCfg: [],
+            userRemoteConfigs: [[
+                url: 'https://github.com/manasagowri/cephci.git'
```

Comment:
The git URL must be the one that is specified in the params

Reply:
Done.